05.wc/wcc.rb

```diff
@@ -0,0 +1,124 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: false
+
+require 'optparse'
+
+WIDTH_SUM_VALUES = 8
+
+# Rubyでつくるwcコマンド
+#   Done: コマンド引数からは、複数ファイルを指定可能にする
+#   Done: パイプラインからも入力可能にする
+#   Done: -lオプションを作る 行数をカウント
+#   Done: -wオプションを作る 単語数をカウント
+#   Done: -cオプションを作る バイト数をカウント
+def main(args)
+  options = {}
+  options[:filepaths] = args[:filepaths] ||= false
+  options[:l]         = args[:l] ||= false
+  options[:w]         = args[:w] ||= false
+  options[:c]         = args[:c] ||= false
+
+  input_params = correct_inputs(options[:filepaths])
+  input_params = count_inputs(input_params)
+  show_result(input_params, options)
+end
+
+def correct_inputs(filepaths)
+  result = []
+  if filepaths.empty?
+    stdin = {}
+    stdin[:filepath] = nil
+    stdin[:body] = readlines.map(&:chomp)
+    result << stdin
```

Comment:
resultに入れずに

```suggestion
    [stdin]
```

を戻り値にしてしまう方法があります。

Reply:
配列扱いにできるんですね。ご指摘のとおり修正しました。